-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2d503c/22615/summary.html
**COMMIT**: ef10c8b17c495af1e443da15475c166e74a33402
**CMSSW**: CMSSW_12_3_X_2022-02-22-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/37046/22615/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @mmusich cms-sw/cmssw#36980
 - @iarspider cms-sw/cmssw#37008

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2d503c/22615/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2d503c/22615/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MomentumDependentPedeLabeler.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/PedeLabeler.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/RunRangeDependentPedeLabeler.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/ZMuMuMassConstraintParameterFinder.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeAlignmentAlgorithm.cc: In member function 'std::pair<unsigned int, unsigned int> MillePedeAlignmentAlgorithm::addReferenceTrajectory(const edm::EventSetup&, const AlignmentAlgorithmBase::EventInfo&, const value_type&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeAlignmentAlgorithm.cc:517:24: error: 'class gbl::GblPoint' has no member named 'numMeasurements'; did you mean 'addMeasurement'?
  517 |           if (itPoint->numMeasurements() >= 1)
      |                        ^~~~~~~~~~~~~~~
      |                        addMeasurement
gmake: *** [tmp/slc7_amd64_gcc10/src/Alignment/MillePedeAlignmentAlgorithm/plugins/AlignmentMillePedeAlignmentAlgorithmAuto/MillePedeAlignmentAlgorithm.cc.o] Error 1
>> Building  edm plugin tmp/slc7_amd64_gcc10/src/Alignment/MillePedeAlignmentAlgorithm/plugins/AlignmentMillePedeAlignmentAlgorithmAuto/libAlignmentMillePedeAlignmentAlgorithmAuto.so
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package RecoTracker/TkDetLayers
>> Entering Package RecoTracker/TkNavigation
>> Entering Package RecoTracker/TkSeedingLayers
>> Entering Package RecoTracker/TransientTrackingRecHit
>> Compile sequence completed for CMSSW CMSSW_12_3_X_2022-02-22-2300
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --ignoreWarning=Wdeprecated-declarations --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/tmp/slc7_amd64_gcc10/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package Alignment/MillePedeAlignmentAlgorithm
Entering library rule at src/Alignment/MillePedeAlignmentAlgorithm/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-22-2300/src/Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc
</pre>

--- a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
+++ b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
@@ -145,6 +145,17 @@ def customiseFor2018Input(process):
     return process
 
 
+
+def customiseFor37046(process):
+    """ Customisation to remove the PrescaleCSVFile parameter from the ParametersetDescription of the L1TGlobalProducer
+     in PR 37046 (https://github.com/cms-sw/cmssw/pull/37046)
+    """
+    for prod in producers_by_type(process, 'L1TGlobalProducer'):
+        if hasattr(prod, 'PrescaleCSVFile'):
+            delattr(prod, 'PrescaleCSVFile')
+    return process
+
+
 # CMSSW version specific customizations
 def customizeHLTforCMSSW(process, menuType="GRun"):
 
@@ -154,5 +165,7 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
 
     # add call to action function in proper order: newest last!
     # process = customiseFor12718(process)
+
+    process = customiseFor37046(process)
 
     return process

